### PR TITLE
Add Snowflake authenticator param to docs

### PR DESCRIPTION
**Pull request by foundinblank**
Add Snowflake `authenticator` details to docs (from #644)

### CODE REVIEW

File: docs/guides/connections.md [hunk 1]
```diff
@@ -109,9 +109,10 @@ A Snowflake connection should be configured as follows:
 connections:
     my_snowflake_connection:
         type: snowflake
-        user: [username]
-        password: [password]
-        account: [account ID]
+        user: [required, username]
```
Comment (GeorgeSittas) [inline docs/guides/connections.md:112]: Is this always required? Doesn't seem to match the field type in https://github.com/TobikoData/sqlmesh/commit/1220e602cea573479e9c70b792bc6a0614415ab4#diff-95f5a476d8642ba1937252ee40000b28f73c881331613e66abd63dac92514416R105.

cc: @eakmanrq

File: docs/integrations/engines.md [hunk 2]
```diff
@@ -180,14 +180,15 @@ sqlmesh_airflow = SQLMeshAirflow(
...
-|-------------|------------------------------|:------:|:--------:|
-| `user`      | The Snowflake username       | string |    Y     |
-| `password`  | The Snowflake password       | string |    Y     |
-| `account`   | The Snowflake account name   | string |    Y     |
-| `warehouse` | The Snowflake warehouse name | string |    N     |
-| `database`  | The Snowflake database name  | string |    N     |
-| `role`      | The Snowflake role name      | string |    N     |
+| Option          | Description                        |  Type  | Required |
+|-----------------|------------------------------------|:------:|:--------:|
+| `user`          | The Snowflake username             | string |    Y     |
```
Comment (GeorgeSittas) [inline docs/integrations/engines.md:185]: Ditto.

**Review by GeorgeSittas — APPROVED** [review]
Looks good, just one small question.

Thanks for updating this!

File: docs/guides/connections.md (hunk 1 above)
Comment (eakmanrq) [reply docs/guides/connections.md:112]: I believe it is required if doing externalbrowser or basic auth but not required for Okta. So there is some nuance here. Maybe just update to say "required if not using Okta"?

**Comment by eakmanrq** [comment]
Thanks for catching this @foundinblank. Just a small comment on the docs for username.

**Pull request merged**